Running tests locally

Hi,

Thinking about starting to write some PRs against this project, I was looking to run the tests locally to confirm everything was clean before starting, and that any changes I make still pass tests. I'm having a difficult time getting the tests to run at all, and am seeing lots of failures that don't appear to be happening when running under Github Actions. Are there some undocumented environmental steps required to make tests run?

- `go test` fails with `t.Parallel called multiple times`. Running the test suite with `env CI=1` appears to fix this:

<details>
  <summary>Stack trace</summary>

```
> go test
--- FAIL: TestPodmanDriver_Pull (0.00s)
panic: testing: t.Parallel called multiple times [recovered]
        panic: testing: t.Parallel called multiple times

goroutine 75 [running]:
testing.tRunner.func1.2({0xdec660, 0x10c3090})
        /home/ben/go/src/testing/testing.go:1209 +0x24e
testing.tRunner.func1()
        /home/ben/go/src/testing/testing.go:1212 +0x218
panic({0xdec660, 0x10c3090})
        /home/ben/go/src/runtime/panic.go:1038 +0x215
testing.(*T).Parallel(0xf6d332)
        /home/ben/go/src/testing/testing.go:1077 +0x298
github.com/hashicorp/nomad-driver-podman.startDestroyInspect(_, {{0x1758ff0, 0x4, 0x4}, {{0x0, 0x0}, {0x0, 0x0}}, {0x0, 0x0, ...}, ...}, ...)
        /home/ben/git/nomad-driver-podman/driver_test.go:1973 +0xf6
github.com/hashicorp/nomad-driver-podman.startDestroyInspectImage(0xc000150ea0, {0xf7929f, 0x10}, {0xf728c0, 0x8})
        /home/ben/git/nomad-driver-podman/driver_test.go:1744 +0x165
github.com/hashicorp/nomad-driver-podman.TestPodmanDriver_Pull(0xc000150b60)
        /home/ben/git/nomad-driver-podman/driver_test.go:1738 +0x138
testing.tRunner(0xc000150ea0, 0xfc1918)
        /home/ben/go/src/testing/testing.go:1259 +0x102
created by testing.(*T).Run
        /home/ben/go/src/testing/testing.go:1306 +0x35a
exit status 2
FAIL    github.com/hashicorp/nomad-driver-podman        0.027s
```
</details>

- Tests appear to expect `podman system service` is running. Running `podman system service --time=0 unix:///run/user/1093000004/podman/podman.sock &` before starting the test suite appears to fix this error

<details>
  <summary>dial unix /run/user/1093000004/podman/podman.sock connect: no such file or directory</summary>

```
> go test
2022-01-05T10:18:55.545Z [ERROR] nomad-driver-podman/driver.go:224: podman: Could not get podman info: error="Get "http://u/v1.0.0/libpod/info": dial unix /run/user/1093000004/podman/podman.sock: connect: no such file or directory"
```
</details>

- On my corporate workstation (which requires a web proxy for internet access, the pull test appears to fail, despite having appropriate env vars setup. Manually pulling `busybox:musl` before running the test suite appears to work around the error

- On my corp system (CentOS 8, podman 3.0.1, selinux disabled), I'm getting three failed tests

<details>
  <summary>Failed tests</summary>

```
--- FAIL: TestPodmanDriver_Tmpfs (5.29s)
    driver_test.go:1203: 
                Error Trace:    driver_test.go:1203
                Error:          "2022-01-05T10:37:16.755501020+00:00 stdout F tmpfs on /alloc type tmpfs (rw,relatime,size=2097152k)\n2022-01-05T10:37:16.755501020+00:00 stdout F tmpfs on /local type tmpfs (rw,relatime,size=2097152k)\n2022-01-05T10:37:16.755501020+00:00 stdout F tmpfs on /secrets type tmpfs (rw,relatime,size=2097152k)\n2022-01-05T10:37:16.755501020+00:00 stdout F /tmpdata1 on tmpfs type rw,nosuid,nodev,relatime,uid=1234,gid=10000 (0)\n2022-01-05T10:37:16.755501020+00:00 stdout F /tmpdata2 on tmpfs type rw,nosuid,nodev,relatime,uid=1234,gid=10000 (0)\n2022-01-05T10:37:16.755501020+00:00 stdout F tmpfs on /dev type tmpfs (rw,nosuid,size=65536k,mode=755,uid=1234,gid=10000)\n2022-01-05T10:37:16.755501020+00:00 stdout F shm on /dev/shm type tmpfs (rw,nosuid,nodev,noexec,relatime,size=64000k,uid=1234,gid=10000)\n2022-01-05T10:37:16.755501020+00:00 stdout F tmpfs on /run/.containerenv type tmpfs (rw,nosuid,nodev,relatime,size=2034096k,mode=700,uid=1234,gid=10000)\n2022-01-05T10:37:16.755501020+00:00 stdout F tmpfs on /run/secrets type tmpfs (rw,nosuid,nodev,relatime,size=2034096k,mode=700,uid=1234,gid=10000)\n2022-01-05T10:37:16.755501020+00:00 stdout F tmpfs on /etc/resolv.conf type tmpfs (rw,nosuid,nodev,relatime,size=2034096k,mode=700,uid=1234,gid=10000)\n2022-01-05T10:37:16.755501020+00:00 stdout F tmpfs on /etc/hosts type tmpfs (rw,nosuid,nodev,relatime,size=2034096k,mode=700,uid=1234,gid=10000)\n2022-01-05T10:37:16.755501020+00:00 stdout F tmpfs on /etc/hostname type tmpfs (rw,nosuid,nodev,relatime,size=2034096k,mode=700,uid=1234,gid=10000)\n2022-01-05T10:37:16.755501020+00:00 stdout F devtmpfs on /dev/null type devtmpfs (rw,nosuid,size=10153300k,nr_inodes=2538325,mode=755)\n2022-01-05T10:37:16.755501020+00:00 stdout F devtmpfs on /dev/random type devtmpfs (rw,nosuid,size=10153300k,nr_inodes=2538325,mode=755)\n2022-01-05T10:37:16.755501020+00:00 stdout F devtmpfs on /dev/full type devtmpfs (rw,nosuid,size=10153300k,nr_inodes=2538325,mode=755)\n2022-01-05T10:37:16.755501020+00:00 stdout F devtmpfs on /dev/tty type devtmpfs (rw,nosuid,size=10153300k,nr_inodes=2538325,mode=755)\n2022-01-05T10:37:16.755501020+00:00 stdout F devtmpfs on /dev/zero type devtmpfs (rw,nosuid,size=10153300k,nr_inodes=2538325,mode=755)\n2022-01-05T10:37:16.755501020+00:00 stdout F devtmpfs on /dev/urandom type devtmpfs (rw,nosuid,size=10153300k,nr_inodes=2538325,mode=755)\n2022-01-05T10:37:16.755501020+00:00 stdout F tmpfs on /proc/acpi type tmpfs (ro,relatime,uid=1234,gid=10000)\n2022-01-05T10:37:16.755501020+00:00 stdout F devtmpfs on /proc/kcore type devtmpfs (rw,nosuid,size=10153300k,nr_inodes=2538325,mode=755)\n2022-01-05T10:37:16.755501020+00:00 stdout F devtmpfs on /proc/keys type devtmpfs (rw,nosuid,size=10153300k,nr_inodes=2538325,mode=755)\n2022-01-05T10:37:16.755501020+00:00 stdout F devtmpfs on /proc/timer_list type devtmpfs (rw,nosuid,size=10153300k,nr_inodes=2538325,mode=755)\n2022-01-05T10:37:16.755501020+00:00 stdout F devtmpfs on /proc/sched_debug type devtmpfs (rw,nosuid,size=10153300k,nr_inodes=2538325,mode=755)\n2022-01-05T10:37:16.755501020+00:00 stdout F tmpfs on /proc/scsi type tmpfs (ro,relatime,uid=1234,gid=10000)\n2022-01-05T10:37:16.755501020+00:00 stdout F tmpfs on /sys/firmware type tmpfs (ro,relatime,uid=1234,gid=10000)\n2022-01-05T10:37:16.755501020+00:00 stdout F tmpfs on /sys/dev/block type tmpfs (ro,relatime,uid=1234,gid=10000)\n" does not contain " on /tmpdata1 type tmpfs "
                Test:           TestPodmanDriver_Tmpfs
--- FAIL: TestPodmanDriver_NetworkModes (21.83s)
    --- FAIL: TestPodmanDriver_NetworkModes/TestPodmanDriver_NetworkModes_mode_bridge (5.48s)
        driver_test.go:1484: 
                Error Trace:    driver_test.go:1484
                Error:          Not equal: 
                                expected: "10.88.0.1"
                                actual  : ""
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -10.88.0.1
                                +
                Test:           TestPodmanDriver_NetworkModes/TestPodmanDriver_NetworkModes_mode_bridge
--- FAIL: TestPodmanDriver_Sysctl (4.25s)
    driver_test.go:1703: 
                Error Trace:    driver_test.go:1703
                Error:          Received unexpected error:
                                rpc error: code = Unknown desc = failed to start task, could not start container: unknown error, status code: 500: {"cause":"OCI runtime attempted to invoke a command that was not found","message":"container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: write sysctl key net.core.somaxconn: open /proc/sys/net/core/somaxconn: no such file or directory: OCI runtime attempted to invoke a command that was not found","response":500}
                Test:           TestPodmanDriver_Sysctl
```
</details>

- On my home system (Fedora 35, podman 3.4.4, selinux enforcing) I'm getting a different and larger set of failed tests

<details>
  <summary>Failed tests</summary>

```
--- FAIL: TestPodmanDriver_logNomad (5.66s)
    driver_test.go:487:
                Error Trace:    driver_test.go:487
                Error:          "" does not contain "ceb424d4-efbf-2733-6df7-814e1f4bf574"
                Test:           TestPodmanDriver_logNomad
                Messages:       stdoutMagic in stdout

--- FAIL: TestPodmanDriver_Hostname (5.66s)
    driver_test.go:535:
                Error Trace:    driver_test.go:535
                Error:          "" does not contain "host_55198663-ee1e-5dfd-ed79-290efd80d3e5"
                Test:           TestPodmanDriver_Hostname

--- FAIL: TestPodmanDriver_Init (6.48s)
    driver_test.go:920:
                Error Trace:    driver_test.go:920
                Error:          "" does not contain "/dev/init"
                Test:           TestPodmanDriver_Init

--- FAIL: TestPodmanDriver_User (5.45s)
    driver_test.go:1036:
                Error Trace:    driver_test.go:1036
                Error:          "" does not contain "www-data"
                Test:           TestPodmanDriver_User

--- FAIL: TestPodmanDriver_Device (6.47s)
    driver_test.go:1077:
                Error Trace:    driver_test.go:1077
                Error:          Should be true
                Test:           TestPodmanDriver_Device

--- FAIL: TestPodmanDriver_Tmpfs (5.94s)
    driver_test.go:1203:
                Error Trace:    driver_test.go:1203
                Error:          "" does not contain " on /tmpdata1 type tmpfs "
                Test:           TestPodmanDriver_Tmpfs

--- FAIL: TestPodmanDriver_Mount (4.39s)
    driver_test.go:1241:
                Error Trace:    driver_test.go:1241
                Error:          Received unexpected error:
                                rpc error: code = Unknown desc = failed to start task, could not start container: unknown error, status code: 500: {"cause":"SELinux relabeling of /tmp is not allowed","message":"SELinux relabeling of /tmp is not allowed","response":500}

--- FAIL: TestPodmanDriver_Dns (5.47s)
    driver_test.go:1412:
                Error Trace:    driver_test.go:1412
                Error:          "" does not contain "nameserver 1.1.1.1"
                Test:           TestPodmanDriver_Dns

--- FAIL: TestPodmanDriver_NetworkMode_Container (10.62s)
    driver_test.go:1559:
                Error Trace:    driver_test.go:1559
                Error:          "" does not contain "127.0.0.1:6748"
                Test:           TestPodmanDriver_NetworkMode_Container

--- FAIL: TestPodmanDriver_NetworkMode_Task (10.71s)
    driver_test.go:1632:
                Error Trace:    driver_test.go:1632
                Error:          "" does not contain "127.0.0.1:6748"
                Test:           TestPodmanDriver_NetworkMode_Task

--- FAIL: TestPodmanDriver_Sysctl (4.44s)
    driver_test.go:1703:
                Error Trace:    driver_test.go:1703
                Error:          Received unexpected error:
                                rpc error: code = Unknown desc = failed to start task, could not start container: unknown error, status code: 500: {"cause":"OCI runtime attempted to invoke a command that was not found","message":"open /proc/sys/net/core/somaxconn: No such file or directory: OCI runtime attempted to invoke a command that was not found","response":500}
                Test:           TestPodmanDriver_Sysctl

FAIL
exit status 1
FAIL    github.com/hashicorp/nomad-driver-podman        315.758s
```
</details>

- I'm also seeing lots of warnings during the test suite run, which I'm not sure if indicates a problem with environment or with the test suite: 

<details>
  <summary>Warnings</summary>

```
2022-01-05T10:27:27.105Z [WARN]  logmon/logmon.go:292: driver_harness.logmon: timed out waiting for read-side of process output pipe to close
2022-01-05T10:27:27.105Z [WARN]  logmon/logmon.go:292: driver_harness.logmon: timed out waiting for read-side of process output pipe to close
```
</details>

Is there a magic incantation or pre-requisite setup required to make these all pass outside of Github Actions?

--- a/driver.go
+++ b/driver.go
@@ -387,6 +387,8 @@ func (d *Driver) StartTask(cfg *drivers.TaskConfig) (*drivers.TaskHandle, *drive
 	if driverConfig.Logging.Driver == "" || driverConfig.Logging.Driver == LOG_DRIVER_NOMAD {
 		// Only modify container loggin path if LogCollection is not disabled
 		if !d.config.DisableLogCollection {
+			createOpts.LogConfiguration.Driver = "k8s-file"
+
 			createOpts.ContainerBasicConfig.LogConfiguration.Path = cfg.StdoutPath
 		}
 	} else if driverConfig.Logging.Driver == LOG_DRIVER_JOURNALD {

